redwood-protocol-guest/src/commonMain/kotlin/app/cash/redwood/protocol/guest/ProtocolBridge.kt

```diff
@@ -64,38 +65,59 @@ public interface ProtocolBridge : EventSink {
 
   public fun emitChanges()
 
+  @RedwoodCodegenApi
   public fun nextId(): Id
 
+  @RedwoodCodegenApi
   public fun appendCreate(
     id: Id,
     tag: WidgetTag,
   )
 
+  @RedwoodCodegenApi
   public fun <T> appendPropertyChange(
     id: Id,
     tag: PropertyTag,
     serializer: KSerializer<T>,
     value: T,
   )
 
+  @RedwoodCodegenApi
   public fun appendPropertyChange(
     id: Id,
     tag: PropertyTag,
     value: Boolean,
   )
 
+  /**
+   * There's a bug in kotlinx.serialization where decodeFromDynamic() is broken for UInt values
+   * larger than MAX_INT. For example, 4294967295 is incorrectly encoded as -1. We work around that
+   * here by special casing that type.
+   *
+   * https://github.com/Kotlin/kotlinx.serialization/issues/2713
+   */
+  @RedwoodCodegenApi
+  public fun appendPropertyChange(
+    id: Id,
+    tag: PropertyTag,
+    value: UInt,
+  )
```

Comment:
This is new! And it’s quite unfortunate!

I don’t think we need to implement something general purpose for other kotlinx.serialization types. UInt is something we’re already using in Cash App, so we’re pinned here.